examples/subcommands.py

```diff
@@ -0,0 +1,80 @@
+import discord
+from discord.ext import commands
+
+description = '''An example bot to showcase subcommands in the discord.ext.commands extension
+module.
+'''
+
+bot = commands.Bot(command_prefix='?', description=description)
+
+# Our `commands.Group` object, which will be our parent command
+# Any checks specified on our group will carry over to subcommands
+# Note that case insensitivity will not carry over from our bot to our subcommands
+@bot.group()
+@commands.guild_only()
+async def greet(ctx):
+    """The group command for our greet commands"""
+    # The parent command is always called even if subcommands are invoked
+    # So we are checking the invoked subcommand with `ctx.invoked_subcommand`
+    if ctx.invoked_subcommand is None:
+        # If none are found we send our help command
+        await ctx.send_help(ctx.command)
+
+# Defining our subcommand as a command of our group
+# We will invoke this command with `?greet hello`
+# We can define additional checks specific to this subcommand
+@greet.command()
+@commands.is_owner()
+async def hello(ctx):
+    """Say hello!"""
+    await ctx.send(f'Hello {ctx.author}!')
+
+# This would be invoked with `?greet goodbye <member>`
+@greet.command()
+async def goodbye(ctx, member: discord.Member):
+    """Say goodbye to someone"""
+    await ctx.send(f'Goodbye {member.author}')
```

Comment:
```suggestion
    await ctx.send(f'Goodbye {member}')
```
I assume this is just supposed to be `member`